src/ctapipe/image/statistics.py

```diff
@@ -92,3 +108,109 @@ def descriptive_statistics(
         skewness=skewness(values, mean=mean, std=std),
         kurtosis=kurtosis(values, mean=mean, std=std),
     )
+
+
+class FeatureAggregator(Component):
+    """Array-event-wise aggregation of image parameters."""
+
+    image_parameters = List(
+        Tuple(Unicode(), Unicode()),
+        default_value=[],
+        help=(
+            "List of 2-Tuples of Strings: ('prefix', 'feature'). "
+            "The image parameter to be aggregated is 'prefix_feature'."
+        ),
+    ).tag(config=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.feature_generator = FeatureGenerator(parent=self)
+        self.quality_query = QualityQuery(parent=self)
+
+    def __call__(self, event: ArrayEventContainer) -> None:
+        """Fill event container with aggregated image parameters."""
+        table = None
+        for tel_id in event.trigger.tels_with_trigger:
+            t = collect_features(event, tel_id)
```

Comment:
It's possible I missed it, but you are collecting features from all telescope in the event, but shouldn't it include only those used in reconstruction? In other words, instead of `tels_with_trigger`, you should have something like `event.dl2.stereo.geometry.telescopes`

Reply:
Right now this is done via a quality query in `aggregate_table`, which should use the same quality criteria as the reconstruction for which the error will be estimated.  But if this is integrated into the train/apply tool for the error estimator, doing it this way would be better, I agree.